frontend/public/kubevirt/components/vm/vm-detail.jsx

```diff
@@ -30,31 +33,45 @@ import { Disk } from '../disk';
 import { NodeLink, LoadingInline } from '../../../components/utils';
 import { menuActions } from './menu-actions';
 
-const VmiEvents = ({obj: vm}) => {
-  const vmi = {
-    kind: VirtualMachineInstanceModel.kind,
-    metadata: {
-      name: vm.metadata.name,
-      namespace: vm.metadata.namespace,
-    },
+const VmEvents = ({ obj: vm }) => {
+  const vmObj = {
+    kind: vm.kind,
+    name: getName(vm),
+    namespace: getNamespace(vm),
+  };
+  const vmFilter = obj => _.isMatch(obj, vmObj);
+  const vmiFilter = obj => _.isMatch(obj, {...vmObj, kind: VirtualMachineInstanceModel.kind});
+  const launcherPodFilter = ({ kind, namespace, name }) =>
+    kind === PodModel.kind && namespace === getNamespace(vm) && name.startsWith(`${VIRT_LAUNCHER_POD_PREFIX}${getName(vm)}-`);
+  const importerPodFilter = ({ kind, namespace, name }) => {
+    // importer pod example importer-<diskName>-<vmname>-<generatedId>
+    // note: diskName may contain '-' which means pod name should have at least 4 parts
+    const podNameSplit = name.split('-');
+    const vmName = podNameSplit.length > 3 ? podNameSplit[podNameSplit.length - 2] : null;
+    if (!vmName) {
+      return false;
+    }
+    return kind === PodModel.kind && namespace === getNamespace(vm) && name.startsWith('importer') && vmName === getName(vm);
```

Comment:
In that case, I think it would be safer to try to match the vm name to the whole string before the last dash, because vm name can also include dashes. There still can be false positives, but the chance is much lower and we don't get false negatives.

Reply:
You are right, Im matching to the whole string before the last dash now.